{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "from numpy import ones\n",
    "from numpy_sugar import ddot\n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "from pandas_plink import read_plink1_bin, Chunk\n",
    "from numpy.linalg import cholesky\n",
    "from numpy_sugar.linalg import economic_svd\n",
    "import xarray as xr\n",
    "from struct_lmm2 import StructLMM2\n",
    "from limix.qc import quantile_gaussianize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<module 'struct_lmm2' from '/hps/nobackup/stegle/users/acuomo/git_repos/struct-lmm2/struct_lmm2/__init__.py'>"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import struct_lmm2 # check that it refers to code in repo\n",
    "struct_lmm2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_files_dir = \"/hps/nobackup/stegle/users/acuomo/all_scripts/struct_LMM2/sc_endodiff/new/input_files/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "chrom = 8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "## this file will map cells to donors, it will also only including donors we have single cell data (a subset of all of HipSci donors)\n",
    "sample_mapping_file = input_files_dir+\"sample_mapping_file.csv\"\n",
    "sample_mapping = pd.read_csv(sample_mapping_file, dtype={\"genotype_individual_id\": str, \"phenotype_sample_id\": str})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unique donors: 126\n"
     ]
    }
   ],
   "source": [
    "## extract unique individuals\n",
    "donors = sample_mapping[\"genotype_individual_id\"].unique()\n",
    "donors.sort()\n",
    "print(\"Number of unique donors: {}\".format(len(donors)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Mapping files: 100%|██████████| 3/3 [06:10<00:00, 123.53s/it]\n"
     ]
    }
   ],
   "source": [
    "## read in genotype file\n",
    "plink_file = \"/hps/nobackup/hipsci/scratch/genotypes/imputed/2017-03-27/Full_Filtered_SNPs_Plink/hipsci.wec.gtarray.HumanCoreExome.imputed_phased.20170327.genotypes.norm.renamed.bed\"\n",
    "# G = read_plink1_bin(plink_file, chunk=Chunk(nsamples=10, nvariants=3))\n",
    "G = read_plink1_bin(plink_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of donors after kinship intersection: 125\n"
     ]
    }
   ],
   "source": [
    "## read in GRM kinship matrix\n",
    "kinship_file = \"/hps/nobackup/hipsci/scratch/genotypes/imputed/2017-03-27/Full_Filtered_SNPs_Plink-F/hipsci.wec.gtarray.HumanCoreExome.imputed_phased.20170327.genotypes.norm.renamed.kinship\"\n",
    "K = pd.read_csv(kinship_file, sep=\"\\t\", index_col=0)\n",
    "assert all(K.columns == K.index)\n",
    "K = xr.DataArray(K.values, dims=[\"sample_0\", \"sample_1\"], coords={\"sample_0\": K.columns, \"sample_1\": K.index})\n",
    "K = K.sortby(\"sample_0\").sortby(\"sample_1\")\n",
    "donors = sorted(set(list(K.sample_0.values)).intersection(donors))\n",
    "print(\"Number of donors after kinship intersection: {}\".format(len(donors)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "## subset to relevant donors\n",
    "K = K.sel(sample_0=donors, sample_1=donors)\n",
    "assert all(K.sample_0 == donors)\n",
    "assert all(K.sample_1 == donors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "## and decompose such as K = L @ L.T\n",
    "L_kinship = cholesky(K.values)\n",
    "L_kinship = xr.DataArray(L_kinship, dims=[\"sample\", \"col\"], coords={\"sample\": K.sample_0.values})\n",
    "assert all(L_kinship.sample.values == K.sample_0.values)\n",
    "del K"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample mapping number of rows BEFORE intersection: 34256\n",
      "Sample mapping number of rows AFTER intersection: 33964\n"
     ]
    }
   ],
   "source": [
    "print(\"Sample mapping number of rows BEFORE intersection: {}\".format(sample_mapping.shape[0]))\n",
    "sample_mapping = sample_mapping[sample_mapping[\"genotype_individual_id\"].isin(donors)]\n",
    "print(\"Sample mapping number of rows AFTER intersection: {}\".format(sample_mapping.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# expand from donors to cells\n",
    "L_expanded = L_kinship.sel(sample=sample_mapping[\"genotype_individual_id\"].values)\n",
    "assert all(L_expanded.sample.values == sample_mapping[\"genotype_individual_id\"].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# environments\n",
    "# cells by PCs (20)\n",
    "E_file = input_files_dir+\"20PCs.csv\"\n",
    "E = pd.read_csv(E_file, index_col = 0)\n",
    "E = xr.DataArray(E.values, dims=[\"cell\", \"pc\"], coords={\"cell\": E.index.values, \"pc\": E.columns.values})\n",
    "E = E.sel(cell=sample_mapping[\"phenotype_sample_id\"].values)\n",
    "assert all(E.cell.values == sample_mapping[\"phenotype_sample_id\"].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "# subselect to only SNPs on right chromosome\n",
    "G_sel = G.where(G.chrom == str(chrom), drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><svg style=\"position: absolute; width: 0; height: 0; overflow: hidden\">\n",
       "<defs>\n",
       "<symbol id=\"icon-database\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M16 0c-8.837 0-16 2.239-16 5v4c0 2.761 7.163 5 16 5s16-2.239 16-5v-4c0-2.761-7.163-5-16-5z\"></path>\n",
       "<path d=\"M16 17c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "<path d=\"M16 26c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "</symbol>\n",
       "<symbol id=\"icon-file-text2\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M28.681 7.159c-0.694-0.947-1.662-2.053-2.724-3.116s-2.169-2.030-3.116-2.724c-1.612-1.182-2.393-1.319-2.841-1.319h-15.5c-1.378 0-2.5 1.121-2.5 2.5v27c0 1.378 1.122 2.5 2.5 2.5h23c1.378 0 2.5-1.122 2.5-2.5v-19.5c0-0.448-0.137-1.23-1.319-2.841zM24.543 5.457c0.959 0.959 1.712 1.825 2.268 2.543h-4.811v-4.811c0.718 0.556 1.584 1.309 2.543 2.268zM28 29.5c0 0.271-0.229 0.5-0.5 0.5h-23c-0.271 0-0.5-0.229-0.5-0.5v-27c0-0.271 0.229-0.5 0.5-0.5 0 0 15.499-0 15.5 0v7c0 0.552 0.448 1 1 1h7v19.5z\"></path>\n",
       "<path d=\"M23 26h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 22h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 18h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "</symbol>\n",
       "</defs>\n",
       "</svg>\n",
       "<style>/* CSS stylesheet for displaying xarray objects in jupyterlab.\n",
       " *\n",
       " */\n",
       "\n",
       ":root {\n",
       "  --xr-font-color0: var(--jp-content-font-color0, rgba(0, 0, 0, 1));\n",
       "  --xr-font-color2: var(--jp-content-font-color2, rgba(0, 0, 0, 0.54));\n",
       "  --xr-font-color3: var(--jp-content-font-color3, rgba(0, 0, 0, 0.38));\n",
       "  --xr-border-color: var(--jp-border-color2, #e0e0e0);\n",
       "  --xr-disabled-color: var(--jp-layout-color3, #bdbdbd);\n",
       "  --xr-background-color: var(--jp-layout-color0, white);\n",
       "  --xr-background-color-row-even: var(--jp-layout-color1, white);\n",
       "  --xr-background-color-row-odd: var(--jp-layout-color2, #eeeeee);\n",
       "}\n",
       "\n",
       "html[theme=dark],\n",
       "body.vscode-dark {\n",
       "  --xr-font-color0: rgba(255, 255, 255, 1);\n",
       "  --xr-font-color2: rgba(255, 255, 255, 0.54);\n",
       "  --xr-font-color3: rgba(255, 255, 255, 0.38);\n",
       "  --xr-border-color: #1F1F1F;\n",
       "  --xr-disabled-color: #515151;\n",
       "  --xr-background-color: #111111;\n",
       "  --xr-background-color-row-even: #111111;\n",
       "  --xr-background-color-row-odd: #313131;\n",
       "}\n",
       "\n",
       ".xr-wrap {\n",
       "  display: block;\n",
       "  min-width: 300px;\n",
       "  max-width: 700px;\n",
       "}\n",
       "\n",
       ".xr-text-repr-fallback {\n",
       "  /* fallback to plain text repr when CSS is not injected (untrusted notebook) */\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-header {\n",
       "  padding-top: 6px;\n",
       "  padding-bottom: 6px;\n",
       "  margin-bottom: 4px;\n",
       "  border-bottom: solid 1px var(--xr-border-color);\n",
       "}\n",
       "\n",
       ".xr-header > div,\n",
       ".xr-header > ul {\n",
       "  display: inline;\n",
       "  margin-top: 0;\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-obj-type,\n",
       ".xr-array-name {\n",
       "  margin-left: 2px;\n",
       "  margin-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-obj-type {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-sections {\n",
       "  padding-left: 0 !important;\n",
       "  display: grid;\n",
       "  grid-template-columns: 150px auto auto 1fr 20px 20px;\n",
       "}\n",
       "\n",
       ".xr-section-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-section-item input {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-section-item input + label {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label {\n",
       "  cursor: pointer;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label:hover {\n",
       "  color: var(--xr-font-color0);\n",
       "}\n",
       "\n",
       ".xr-section-summary {\n",
       "  grid-column: 1;\n",
       "  color: var(--xr-font-color2);\n",
       "  font-weight: 500;\n",
       "}\n",
       "\n",
       ".xr-section-summary > span {\n",
       "  display: inline-block;\n",
       "  padding-left: 0.5em;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in + label:before {\n",
       "  display: inline-block;\n",
       "  content: '►';\n",
       "  font-size: 11px;\n",
       "  width: 15px;\n",
       "  text-align: center;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label:before {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label:before {\n",
       "  content: '▼';\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label > span {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-section-summary,\n",
       ".xr-section-inline-details {\n",
       "  padding-top: 4px;\n",
       "  padding-bottom: 4px;\n",
       "}\n",
       "\n",
       ".xr-section-inline-details {\n",
       "  grid-column: 2 / -1;\n",
       "}\n",
       "\n",
       ".xr-section-details {\n",
       "  display: none;\n",
       "  grid-column: 1 / -1;\n",
       "  margin-bottom: 5px;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked ~ .xr-section-details {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-array-wrap {\n",
       "  grid-column: 1 / -1;\n",
       "  display: grid;\n",
       "  grid-template-columns: 20px auto;\n",
       "}\n",
       "\n",
       ".xr-array-wrap > label {\n",
       "  grid-column: 1;\n",
       "  vertical-align: top;\n",
       "}\n",
       "\n",
       ".xr-preview {\n",
       "  color: var(--xr-font-color3);\n",
       "}\n",
       "\n",
       ".xr-array-preview,\n",
       ".xr-array-data {\n",
       "  padding: 0 5px !important;\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-array-data,\n",
       ".xr-array-in:checked ~ .xr-array-preview {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-array-in:checked ~ .xr-array-data,\n",
       ".xr-array-preview {\n",
       "  display: inline-block;\n",
       "}\n",
       "\n",
       ".xr-dim-list {\n",
       "  display: inline-block !important;\n",
       "  list-style: none;\n",
       "  padding: 0 !important;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list li {\n",
       "  display: inline-block;\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list:before {\n",
       "  content: '(';\n",
       "}\n",
       "\n",
       ".xr-dim-list:after {\n",
       "  content: ')';\n",
       "}\n",
       "\n",
       ".xr-dim-list li:not(:last-child):after {\n",
       "  content: ',';\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-has-index {\n",
       "  font-weight: bold;\n",
       "}\n",
       "\n",
       ".xr-var-list,\n",
       ".xr-var-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-var-item > div,\n",
       ".xr-var-item label,\n",
       ".xr-var-item > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-even);\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-var-item > .xr-var-name:hover span {\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-var-list > li:nth-child(odd) > div,\n",
       ".xr-var-list > li:nth-child(odd) > label,\n",
       ".xr-var-list > li:nth-child(odd) > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-odd);\n",
       "}\n",
       "\n",
       ".xr-var-name {\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-var-dims {\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-var-dtype {\n",
       "  grid-column: 3;\n",
       "  text-align: right;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-var-preview {\n",
       "  grid-column: 4;\n",
       "}\n",
       "\n",
       ".xr-var-name,\n",
       ".xr-var-dims,\n",
       ".xr-var-dtype,\n",
       ".xr-preview,\n",
       ".xr-attrs dt {\n",
       "  white-space: nowrap;\n",
       "  overflow: hidden;\n",
       "  text-overflow: ellipsis;\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-var-name:hover,\n",
       ".xr-var-dims:hover,\n",
       ".xr-var-dtype:hover,\n",
       ".xr-attrs dt:hover {\n",
       "  overflow: visible;\n",
       "  width: auto;\n",
       "  z-index: 1;\n",
       "}\n",
       "\n",
       ".xr-var-attrs,\n",
       ".xr-var-data {\n",
       "  display: none;\n",
       "  background-color: var(--xr-background-color) !important;\n",
       "  padding-bottom: 5px !important;\n",
       "}\n",
       "\n",
       ".xr-var-attrs-in:checked ~ .xr-var-attrs,\n",
       ".xr-var-data-in:checked ~ .xr-var-data {\n",
       "  display: block;\n",
       "}\n",
       "\n",
       ".xr-var-data > table {\n",
       "  float: right;\n",
       "}\n",
       "\n",
       ".xr-var-name span,\n",
       ".xr-var-data,\n",
       ".xr-attrs {\n",
       "  padding-left: 25px !important;\n",
       "}\n",
       "\n",
       ".xr-attrs,\n",
       ".xr-var-attrs,\n",
       ".xr-var-data {\n",
       "  grid-column: 1 / -1;\n",
       "}\n",
       "\n",
       "dl.xr-attrs {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  display: grid;\n",
       "  grid-template-columns: 125px auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt,\n",
       ".xr-attrs dd {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  float: left;\n",
       "  padding-right: 10px;\n",
       "  width: auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt {\n",
       "  font-weight: normal;\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-attrs dt:hover span {\n",
       "  display: inline-block;\n",
       "  background: var(--xr-background-color);\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-attrs dd {\n",
       "  grid-column: 2;\n",
       "  white-space: pre-wrap;\n",
       "  word-break: break-all;\n",
       "}\n",
       "\n",
       ".xr-icon-database,\n",
       ".xr-icon-file-text2 {\n",
       "  display: inline-block;\n",
       "  vertical-align: middle;\n",
       "  width: 1em;\n",
       "  height: 1.5em !important;\n",
       "  stroke-width: 0;\n",
       "  stroke: currentColor;\n",
       "  fill: currentColor;\n",
       "}\n",
       "</style><pre class='xr-text-repr-fallback'>&lt;xarray.DataArray &#x27;genotype&#x27; (sample: 1610, variant: 594310)&gt;\n",
       "dask.array&lt;where, shape=(1610, 594310), dtype=float32, chunksize=(1024, 1024), chunktype=numpy.ndarray&gt;\n",
       "Coordinates:\n",
       "  * sample   (sample) object &#x27;HPSI0216i-tony_5&#x27; ... &#x27;HPSI0416pf-cihd&#x27;\n",
       "  * variant  (variant) object &#x27;8_8_66822_T_C&#x27; ... &#x27;8_8_146301235_C_G&#x27;\n",
       "    fid      (sample) object &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; ... &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27;\n",
       "    iid      (sample) object &#x27;HPSI0216i-tony_5&#x27; ... &#x27;HPSI0416pf-cihd&#x27;\n",
       "    father   (sample) object &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; ... &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27;\n",
       "    mother   (sample) object &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; ... &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27;\n",
       "    gender   (sample) object &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; ... &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27;\n",
       "    trait    (sample) object &#x27;-9&#x27; &#x27;-9&#x27; &#x27;-9&#x27; &#x27;-9&#x27; &#x27;-9&#x27; ... &#x27;-9&#x27; &#x27;-9&#x27; &#x27;-9&#x27; &#x27;-9&#x27;\n",
       "    chrom    (variant) object &#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27; ... &#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27;\n",
       "    snp      (variant) object &#x27;8_66822_T_C&#x27; &#x27;8_91936_A_G&#x27; ... &#x27;8_146301235_C_G&#x27;\n",
       "    cm       (variant) float64 0.0 0.0 0.0 0.0 0.0 0.0 ... 0.0 0.0 0.0 0.0 0.0\n",
       "    pos      (variant) int32 66822 91936 127184 ... 146301227 146301235\n",
       "    a0       (variant) object &#x27;T&#x27; &#x27;A&#x27; &#x27;G&#x27; &#x27;T&#x27; &#x27;C&#x27; &#x27;A&#x27; ... &#x27;G&#x27; &#x27;G&#x27; &#x27;G&#x27; &#x27;A&#x27; &#x27;C&#x27;\n",
       "    a1       (variant) object &#x27;C&#x27; &#x27;G&#x27; &#x27;C&#x27; &#x27;C&#x27; &#x27;G&#x27; &#x27;G&#x27; ... &#x27;A&#x27; &#x27;A&#x27; &#x27;A&#x27; &#x27;G&#x27; &#x27;G&#x27;</pre><div class='xr-wrap' hidden><div class='xr-header'><div class='xr-obj-type'>xarray.DataArray</div><div class='xr-array-name'>'genotype'</div><ul class='xr-dim-list'><li><span class='xr-has-index'>sample</span>: 1610</li><li><span class='xr-has-index'>variant</span>: 594310</li></ul></div><ul class='xr-sections'><li class='xr-section-item'><div class='xr-array-wrap'><input id='section-73c8575d-3d7d-433d-b134-1467fa20a3e8' class='xr-array-in' type='checkbox' checked><label for='section-73c8575d-3d7d-433d-b134-1467fa20a3e8' title='Show/hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-array-preview xr-preview'><span>dask.array&lt;chunksize=(1024, 145), meta=np.ndarray&gt;</span></div><div class='xr-array-data'><table>\n",
       "<tr>\n",
       "<td>\n",
       "<table>\n",
       "  <thead>\n",
       "    <tr><td> </td><th> Array </th><th> Chunk </th></tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr><th> Bytes </th><td> 3.83 GB </td> <td> 4.19 MB </td></tr>\n",
       "    <tr><th> Shape </th><td> (1610, 594310) </td> <td> (1024, 1024) </td></tr>\n",
       "    <tr><th> Count </th><td> 105693 Tasks </td><td> 1164 Chunks </td></tr>\n",
       "    <tr><th> Type </th><td> float32 </td><td> numpy.ndarray </td></tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</td>\n",
       "<td>\n",
       "<svg width=\"170\" height=\"75\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"16\" x2=\"120\" y2=\"16\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"6\" y1=\"0\" x2=\"6\" y2=\"25\" />\n",
       "  <line x1=\"12\" y1=\"0\" x2=\"12\" y2=\"25\" />\n",
       "  <line x1=\"18\" y1=\"0\" x2=\"18\" y2=\"25\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"25\" />\n",
       "  <line x1=\"31\" y1=\"0\" x2=\"31\" y2=\"25\" />\n",
       "  <line x1=\"37\" y1=\"0\" x2=\"37\" y2=\"25\" />\n",
       "  <line x1=\"44\" y1=\"0\" x2=\"44\" y2=\"25\" />\n",
       "  <line x1=\"50\" y1=\"0\" x2=\"50\" y2=\"25\" />\n",
       "  <line x1=\"56\" y1=\"0\" x2=\"56\" y2=\"25\" />\n",
       "  <line x1=\"63\" y1=\"0\" x2=\"63\" y2=\"25\" />\n",
       "  <line x1=\"69\" y1=\"0\" x2=\"69\" y2=\"25\" />\n",
       "  <line x1=\"75\" y1=\"0\" x2=\"75\" y2=\"25\" />\n",
       "  <line x1=\"82\" y1=\"0\" x2=\"82\" y2=\"25\" />\n",
       "  <line x1=\"88\" y1=\"0\" x2=\"88\" y2=\"25\" />\n",
       "  <line x1=\"94\" y1=\"0\" x2=\"94\" y2=\"25\" />\n",
       "  <line x1=\"101\" y1=\"0\" x2=\"101\" y2=\"25\" />\n",
       "  <line x1=\"107\" y1=\"0\" x2=\"107\" y2=\"25\" />\n",
       "  <line x1=\"113\" y1=\"0\" x2=\"113\" y2=\"25\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"45.412617\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >594310</text>\n",
       "  <text x=\"140.000000\" y=\"12.706308\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(-90,140.000000,12.706308)\">1610</text>\n",
       "</svg>\n",
       "</td>\n",
       "</tr>\n",
       "</table></div></div></li><li class='xr-section-item'><input id='section-77727a90-46c4-4ab6-ba58-734aa18855e7' class='xr-section-summary-in' type='checkbox'  checked><label for='section-77727a90-46c4-4ab6-ba58-734aa18855e7' class='xr-section-summary' >Coordinates: <span>(14)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'><li class='xr-var-item'><div class='xr-var-name'><span class='xr-has-index'>sample</span></div><div class='xr-var-dims'>(sample)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;HPSI0216i-tony_5&#x27; ... &#x27;HPSI0416...</div><input id='attrs-1a8badf3-caee-4b88-a7b3-f817a22da79f' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-1a8badf3-caee-4b88-a7b3-f817a22da79f' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-b99064c2-0819-40a6-a4e7-1cd899f2d6c9' class='xr-var-data-in' type='checkbox'><label for='data-b99064c2-0819-40a6-a4e7-1cd899f2d6c9' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;HPSI0216i-tony_5&#x27;, &#x27;HPSI0216pf-tony&#x27;, &#x27;HPSI0316i-xaqm_4&#x27;, ...,\n",
       "       &#x27;HPSI0516pf-eosr&#x27;, &#x27;HPSI0616pf-xuvy&#x27;, &#x27;HPSI0416pf-cihd&#x27;], dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span class='xr-has-index'>variant</span></div><div class='xr-var-dims'>(variant)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;8_8_66822_T_C&#x27; ... &#x27;8_8_1463012...</div><input id='attrs-771462c3-2bea-428e-a1f4-e2815ccdec44' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-771462c3-2bea-428e-a1f4-e2815ccdec44' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-e4a3ef38-1e03-4465-a1df-b25ac0cf30f9' class='xr-var-data-in' type='checkbox'><label for='data-e4a3ef38-1e03-4465-a1df-b25ac0cf30f9' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;8_8_66822_T_C&#x27;, &#x27;8_8_91936_A_G&#x27;, &#x27;8_8_127184_G_C&#x27;, ...,\n",
       "       &#x27;8_8_146301221_G_A&#x27;, &#x27;8_8_146301227_A_G&#x27;, &#x27;8_8_146301235_C_G&#x27;],\n",
       "      dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>fid</span></div><div class='xr-var-dims'>(sample)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; ... &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27;</div><input id='attrs-f5053c6b-0658-4d83-a472-5fa7f6d95d99' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-f5053c6b-0658-4d83-a472-5fa7f6d95d99' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-fcc475b9-00d3-4b86-a1e6-806bfb391c3f' class='xr-var-data-in' type='checkbox'><label for='data-fcc475b9-00d3-4b86-a1e6-806bfb391c3f' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;0&#x27;, &#x27;0&#x27;, &#x27;0&#x27;, ..., &#x27;0&#x27;, &#x27;0&#x27;, &#x27;0&#x27;], dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>iid</span></div><div class='xr-var-dims'>(sample)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;HPSI0216i-tony_5&#x27; ... &#x27;HPSI0416...</div><input id='attrs-08fe2a45-6971-45e6-b734-93f2cd72fe7e' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-08fe2a45-6971-45e6-b734-93f2cd72fe7e' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-30b5810e-45fe-4cfc-ba71-9b2c5d19b8b2' class='xr-var-data-in' type='checkbox'><label for='data-30b5810e-45fe-4cfc-ba71-9b2c5d19b8b2' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;HPSI0216i-tony_5&#x27;, &#x27;HPSI0216pf-tony&#x27;, &#x27;HPSI0316i-xaqm_4&#x27;, ...,\n",
       "       &#x27;HPSI0516pf-eosr&#x27;, &#x27;HPSI0616pf-xuvy&#x27;, &#x27;HPSI0416pf-cihd&#x27;],\n",
       "      dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>father</span></div><div class='xr-var-dims'>(sample)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; ... &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27;</div><input id='attrs-a104032e-7ce6-481e-81d0-cbd9cafa6dda' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-a104032e-7ce6-481e-81d0-cbd9cafa6dda' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-c0e87d36-1efb-4a0e-9e78-8cc969cae589' class='xr-var-data-in' type='checkbox'><label for='data-c0e87d36-1efb-4a0e-9e78-8cc969cae589' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;0&#x27;, &#x27;0&#x27;, &#x27;0&#x27;, ..., &#x27;0&#x27;, &#x27;0&#x27;, &#x27;0&#x27;], dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>mother</span></div><div class='xr-var-dims'>(sample)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; ... &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27;</div><input id='attrs-0e353e75-56b9-449f-a1ae-b0198daafa25' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-0e353e75-56b9-449f-a1ae-b0198daafa25' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-9ce0c820-1ef8-40a3-9c2d-daf507df347a' class='xr-var-data-in' type='checkbox'><label for='data-9ce0c820-1ef8-40a3-9c2d-daf507df347a' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;0&#x27;, &#x27;0&#x27;, &#x27;0&#x27;, ..., &#x27;0&#x27;, &#x27;0&#x27;, &#x27;0&#x27;], dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>gender</span></div><div class='xr-var-dims'>(sample)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; ... &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27; &#x27;0&#x27;</div><input id='attrs-7ae00d76-76a6-44ac-bf9f-66ece9adda94' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-7ae00d76-76a6-44ac-bf9f-66ece9adda94' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-8bb7be8c-acfa-4ea0-b5e9-97702b80dfa6' class='xr-var-data-in' type='checkbox'><label for='data-8bb7be8c-acfa-4ea0-b5e9-97702b80dfa6' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;0&#x27;, &#x27;0&#x27;, &#x27;0&#x27;, ..., &#x27;0&#x27;, &#x27;0&#x27;, &#x27;0&#x27;], dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>trait</span></div><div class='xr-var-dims'>(sample)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;-9&#x27; &#x27;-9&#x27; &#x27;-9&#x27; ... &#x27;-9&#x27; &#x27;-9&#x27; &#x27;-9&#x27;</div><input id='attrs-b502d323-267c-41bf-8bf7-7b5ffc400049' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-b502d323-267c-41bf-8bf7-7b5ffc400049' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-cea0b26b-7a1a-4d04-bb56-6f00785fae4a' class='xr-var-data-in' type='checkbox'><label for='data-cea0b26b-7a1a-4d04-bb56-6f00785fae4a' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;-9&#x27;, &#x27;-9&#x27;, &#x27;-9&#x27;, ..., &#x27;-9&#x27;, &#x27;-9&#x27;, &#x27;-9&#x27;], dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>chrom</span></div><div class='xr-var-dims'>(variant)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27; ... &#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27; &#x27;8&#x27;</div><input id='attrs-503fd72e-7fd2-4497-896f-949d7312f4f6' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-503fd72e-7fd2-4497-896f-949d7312f4f6' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-e9a4bc35-6c7f-420c-b07f-a2567cec8c13' class='xr-var-data-in' type='checkbox'><label for='data-e9a4bc35-6c7f-420c-b07f-a2567cec8c13' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;8&#x27;, &#x27;8&#x27;, &#x27;8&#x27;, ..., &#x27;8&#x27;, &#x27;8&#x27;, &#x27;8&#x27;], dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>snp</span></div><div class='xr-var-dims'>(variant)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;8_66822_T_C&#x27; ... &#x27;8_146301235_C_G&#x27;</div><input id='attrs-3e037ce3-2224-40f3-8969-67578c9cfeaa' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-3e037ce3-2224-40f3-8969-67578c9cfeaa' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-b3aa3602-6dd5-46a1-99c2-282e10c1e1fa' class='xr-var-data-in' type='checkbox'><label for='data-b3aa3602-6dd5-46a1-99c2-282e10c1e1fa' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;8_66822_T_C&#x27;, &#x27;8_91936_A_G&#x27;, &#x27;8_127184_G_C&#x27;, ...,\n",
       "       &#x27;8_146301221_G_A&#x27;, &#x27;8_146301227_A_G&#x27;, &#x27;8_146301235_C_G&#x27;],\n",
       "      dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>cm</span></div><div class='xr-var-dims'>(variant)</div><div class='xr-var-dtype'>float64</div><div class='xr-var-preview xr-preview'>0.0 0.0 0.0 0.0 ... 0.0 0.0 0.0 0.0</div><input id='attrs-50f0954b-aec6-4235-9883-4916e2c826bd' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-50f0954b-aec6-4235-9883-4916e2c826bd' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-7ad54920-e127-40ad-a367-1d0f48e8f2ad' class='xr-var-data-in' type='checkbox'><label for='data-7ad54920-e127-40ad-a367-1d0f48e8f2ad' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([0., 0., 0., ..., 0., 0., 0.])</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>pos</span></div><div class='xr-var-dims'>(variant)</div><div class='xr-var-dtype'>int32</div><div class='xr-var-preview xr-preview'>66822 91936 ... 146301227 146301235</div><input id='attrs-9539dc4d-b701-4761-9080-bab34b90278c' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-9539dc4d-b701-4761-9080-bab34b90278c' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-48fef72a-6114-470d-9745-d1fa65dd11c9' class='xr-var-data-in' type='checkbox'><label for='data-48fef72a-6114-470d-9745-d1fa65dd11c9' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([    66822,     91936,    127184, ..., 146301221, 146301227,\n",
       "       146301235], dtype=int32)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>a0</span></div><div class='xr-var-dims'>(variant)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;T&#x27; &#x27;A&#x27; &#x27;G&#x27; &#x27;T&#x27; ... &#x27;G&#x27; &#x27;G&#x27; &#x27;A&#x27; &#x27;C&#x27;</div><input id='attrs-b2ccac86-0f89-4a88-a871-3ab65d3717cb' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-b2ccac86-0f89-4a88-a871-3ab65d3717cb' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-f06d789d-02bf-4582-929f-49cbfaf8678e' class='xr-var-data-in' type='checkbox'><label for='data-f06d789d-02bf-4582-929f-49cbfaf8678e' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;T&#x27;, &#x27;A&#x27;, &#x27;G&#x27;, ..., &#x27;G&#x27;, &#x27;A&#x27;, &#x27;C&#x27;], dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span>a1</span></div><div class='xr-var-dims'>(variant)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;C&#x27; &#x27;G&#x27; &#x27;C&#x27; &#x27;C&#x27; ... &#x27;A&#x27; &#x27;A&#x27; &#x27;G&#x27; &#x27;G&#x27;</div><input id='attrs-671d083c-40be-4a31-833f-c6b7f97e173d' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-671d083c-40be-4a31-833f-c6b7f97e173d' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-e823e1f1-e81a-4cf7-9ea2-1a81f599e4ce' class='xr-var-data-in' type='checkbox'><label for='data-e823e1f1-e81a-4cf7-9ea2-1a81f599e4ce' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;C&#x27;, &#x27;G&#x27;, &#x27;C&#x27;, ..., &#x27;A&#x27;, &#x27;G&#x27;, &#x27;G&#x27;], dtype=object)</pre></div></li></ul></div></li><li class='xr-section-item'><input id='section-e35c26ff-f7d7-4acf-bc69-26691f52b5d3' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-e35c26ff-f7d7-4acf-bc69-26691f52b5d3' class='xr-section-summary'  title='Expand/collapse section'>Attributes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><dl class='xr-attrs'></dl></div></li></ul></div></div>"
      ],
      "text/plain": [
       "<xarray.DataArray 'genotype' (sample: 1610, variant: 594310)>\n",
       "dask.array<where, shape=(1610, 594310), dtype=float32, chunksize=(1024, 1024), chunktype=numpy.ndarray>\n",
       "Coordinates:\n",
       "  * sample   (sample) object 'HPSI0216i-tony_5' ... 'HPSI0416pf-cihd'\n",
       "  * variant  (variant) object '8_8_66822_T_C' ... '8_8_146301235_C_G'\n",
       "    fid      (sample) object '0' '0' '0' '0' '0' '0' ... '0' '0' '0' '0' '0' '0'\n",
       "    iid      (sample) object 'HPSI0216i-tony_5' ... 'HPSI0416pf-cihd'\n",
       "    father   (sample) object '0' '0' '0' '0' '0' '0' ... '0' '0' '0' '0' '0' '0'\n",
       "    mother   (sample) object '0' '0' '0' '0' '0' '0' ... '0' '0' '0' '0' '0' '0'\n",
       "    gender   (sample) object '0' '0' '0' '0' '0' '0' ... '0' '0' '0' '0' '0' '0'\n",
       "    trait    (sample) object '-9' '-9' '-9' '-9' '-9' ... '-9' '-9' '-9' '-9'\n",
       "    chrom    (variant) object '8' '8' '8' '8' '8' '8' ... '8' '8' '8' '8' '8'\n",
       "    snp      (variant) object '8_66822_T_C' '8_91936_A_G' ... '8_146301235_C_G'\n",
       "    cm       (variant) float64 0.0 0.0 0.0 0.0 0.0 0.0 ... 0.0 0.0 0.0 0.0 0.0\n",
       "    pos      (variant) int32 66822 91936 127184 ... 146301227 146301235\n",
       "    a0       (variant) object 'T' 'A' 'G' 'T' 'C' 'A' ... 'G' 'G' 'G' 'A' 'C'\n",
       "    a1       (variant) object 'C' 'G' 'C' 'C' 'G' 'G' ... 'A' 'A' 'A' 'G' 'G'"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "G_sel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/nfs/software/stegle/users/acuomo/conda-envs/struct-lmm2/lib/python3.8/site-packages/xarray/core/indexing.py:1369: PerformanceWarning: Slicing with an out-of-order index is generating 2383 times more chunks\n",
      "  return self.array[key]\n",
      "/nfs/software/stegle/users/acuomo/conda-envs/struct-lmm2/lib/python3.8/site-packages/xarray/core/indexing.py:1369: PerformanceWarning: Slicing is producing a large chunk. To accept the large\n",
      "chunk and silence this warning, set the option\n",
      "    >>> with dask.config.set(**{'array.slicing.split_large_chunks': False}):\n",
      "    ...     array[indexer]\n",
      "\n",
      "To avoid creating the large chunks, set the option\n",
      "    >>> with dask.config.set(**{'array.slicing.split_large_chunks': True}):\n",
      "    ...     array[indexer]\n",
      "  return self.array[key]\n"
     ]
    }
   ],
   "source": [
    "G_exp = G_sel.sel(sample=sample_mapping[\"genotype_individual_id\"].values)\n",
    "assert all(L_expanded.sample.values == G_exp.sample.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_factors = 10\n",
    "E = E.values[:,0:n_factors]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get eigendecomposition of EEt\n",
    "[U, S, _] = economic_svd(E)\n",
    "us = U * S\n",
    "# get decomposition of K*EEt\n",
    "Ls = [ddot(us[:,i], L_expanded) for i in range(us.shape[1])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(33964, 125)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Ls[1].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>21843_1#10</th>\n",
       "      <th>21843_1#100</th>\n",
       "      <th>21843_1#101</th>\n",
       "      <th>21843_1#102</th>\n",
       "      <th>21843_1#103</th>\n",
       "      <th>21843_1#105</th>\n",
       "      <th>21843_1#106</th>\n",
       "      <th>21843_1#107</th>\n",
       "      <th>21843_1#108</th>\n",
       "      <th>21843_1#109</th>\n",
       "      <th>...</th>\n",
       "      <th>24539_8#88</th>\n",
       "      <th>24539_8#89</th>\n",
       "      <th>24539_8#90</th>\n",
       "      <th>24539_8#91</th>\n",
       "      <th>24539_8#92</th>\n",
       "      <th>24539_8#93</th>\n",
       "      <th>24539_8#94</th>\n",
       "      <th>24539_8#95</th>\n",
       "      <th>24539_8#97</th>\n",
       "      <th>24539_8#98</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003_TSPAN6</th>\n",
       "      <td>5.520777</td>\n",
       "      <td>6.456208</td>\n",
       "      <td>5.878671</td>\n",
       "      <td>4.860824</td>\n",
       "      <td>5.903640</td>\n",
       "      <td>4.513537</td>\n",
       "      <td>6.401983</td>\n",
       "      <td>5.909216</td>\n",
       "      <td>5.366645</td>\n",
       "      <td>3.228852</td>\n",
       "      <td>...</td>\n",
       "      <td>5.841814</td>\n",
       "      <td>6.104105</td>\n",
       "      <td>6.275649</td>\n",
       "      <td>7.029407</td>\n",
       "      <td>5.806978</td>\n",
       "      <td>6.199875</td>\n",
       "      <td>7.014180</td>\n",
       "      <td>6.228476</td>\n",
       "      <td>6.217161</td>\n",
       "      <td>6.034232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419_DPM1</th>\n",
       "      <td>5.392461</td>\n",
       "      <td>6.065923</td>\n",
       "      <td>6.838769</td>\n",
       "      <td>6.614268</td>\n",
       "      <td>6.512403</td>\n",
       "      <td>5.527439</td>\n",
       "      <td>6.525591</td>\n",
       "      <td>6.381135</td>\n",
       "      <td>6.157296</td>\n",
       "      <td>6.248478</td>\n",
       "      <td>...</td>\n",
       "      <td>6.543807</td>\n",
       "      <td>6.369119</td>\n",
       "      <td>7.185421</td>\n",
       "      <td>6.337047</td>\n",
       "      <td>6.162437</td>\n",
       "      <td>5.885993</td>\n",
       "      <td>7.431358</td>\n",
       "      <td>7.013124</td>\n",
       "      <td>4.851771</td>\n",
       "      <td>4.937248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457_SCYL3</th>\n",
       "      <td>0.000174</td>\n",
       "      <td>0.352597</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.825955</td>\n",
       "      <td>2.201697</td>\n",
       "      <td>0.262446</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.506837</td>\n",
       "      <td>0.283516</td>\n",
       "      <td>3.241977</td>\n",
       "      <td>...</td>\n",
       "      <td>0.017386</td>\n",
       "      <td>0.949668</td>\n",
       "      <td>0.035526</td>\n",
       "      <td>0.032044</td>\n",
       "      <td>1.773369</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.108025</td>\n",
       "      <td>1.756339</td>\n",
       "      <td>2.492943</td>\n",
       "      <td>1.363441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460_C1orf112</th>\n",
       "      <td>1.471928</td>\n",
       "      <td>4.536968</td>\n",
       "      <td>4.318528</td>\n",
       "      <td>5.373009</td>\n",
       "      <td>4.636175</td>\n",
       "      <td>4.225468</td>\n",
       "      <td>0.409785</td>\n",
       "      <td>3.668277</td>\n",
       "      <td>3.057933</td>\n",
       "      <td>3.154891</td>\n",
       "      <td>...</td>\n",
       "      <td>4.720967</td>\n",
       "      <td>3.791536</td>\n",
       "      <td>2.696476</td>\n",
       "      <td>4.227515</td>\n",
       "      <td>4.243689</td>\n",
       "      <td>3.227508</td>\n",
       "      <td>2.621121</td>\n",
       "      <td>3.950978</td>\n",
       "      <td>3.926914</td>\n",
       "      <td>4.211904</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000001036_FUCA2</th>\n",
       "      <td>2.908802</td>\n",
       "      <td>3.867327</td>\n",
       "      <td>3.321747</td>\n",
       "      <td>3.736476</td>\n",
       "      <td>4.917576</td>\n",
       "      <td>2.456866</td>\n",
       "      <td>0.577839</td>\n",
       "      <td>4.777404</td>\n",
       "      <td>2.873857</td>\n",
       "      <td>2.536708</td>\n",
       "      <td>...</td>\n",
       "      <td>3.070608</td>\n",
       "      <td>4.491643</td>\n",
       "      <td>4.206249</td>\n",
       "      <td>3.695005</td>\n",
       "      <td>2.652845</td>\n",
       "      <td>4.221847</td>\n",
       "      <td>3.188030</td>\n",
       "      <td>4.741496</td>\n",
       "      <td>3.872743</td>\n",
       "      <td>4.374577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000271043_MTRNR2L2</th>\n",
       "      <td>8.993125</td>\n",
       "      <td>10.310311</td>\n",
       "      <td>10.112632</td>\n",
       "      <td>9.380978</td>\n",
       "      <td>9.511378</td>\n",
       "      <td>7.531314</td>\n",
       "      <td>9.565223</td>\n",
       "      <td>9.659486</td>\n",
       "      <td>7.748536</td>\n",
       "      <td>10.087547</td>\n",
       "      <td>...</td>\n",
       "      <td>5.273407</td>\n",
       "      <td>5.649628</td>\n",
       "      <td>5.465876</td>\n",
       "      <td>5.388675</td>\n",
       "      <td>5.636233</td>\n",
       "      <td>1.709041</td>\n",
       "      <td>7.568684</td>\n",
       "      <td>6.345048</td>\n",
       "      <td>7.577899</td>\n",
       "      <td>7.012576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000271303_SRXN1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.390075</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.388986</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.061808</td>\n",
       "      <td>1.707139</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.637493</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.284224</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000272047_GTF2H5</th>\n",
       "      <td>1.397666</td>\n",
       "      <td>1.399811</td>\n",
       "      <td>2.421133</td>\n",
       "      <td>2.250520</td>\n",
       "      <td>2.633359</td>\n",
       "      <td>2.078498</td>\n",
       "      <td>1.513069</td>\n",
       "      <td>2.972922</td>\n",
       "      <td>2.012249</td>\n",
       "      <td>1.675936</td>\n",
       "      <td>...</td>\n",
       "      <td>2.316120</td>\n",
       "      <td>2.299498</td>\n",
       "      <td>2.392277</td>\n",
       "      <td>1.653269</td>\n",
       "      <td>2.254969</td>\n",
       "      <td>3.208027</td>\n",
       "      <td>4.137374</td>\n",
       "      <td>1.847985</td>\n",
       "      <td>2.144427</td>\n",
       "      <td>2.105545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000272325_NUDT3</th>\n",
       "      <td>0.716015</td>\n",
       "      <td>1.857059</td>\n",
       "      <td>2.499615</td>\n",
       "      <td>2.615108</td>\n",
       "      <td>1.682406</td>\n",
       "      <td>2.295494</td>\n",
       "      <td>2.040256</td>\n",
       "      <td>1.896238</td>\n",
       "      <td>1.236113</td>\n",
       "      <td>1.544098</td>\n",
       "      <td>...</td>\n",
       "      <td>1.210306</td>\n",
       "      <td>0.538546</td>\n",
       "      <td>1.002614</td>\n",
       "      <td>1.901335</td>\n",
       "      <td>1.037860</td>\n",
       "      <td>2.139444</td>\n",
       "      <td>0.852408</td>\n",
       "      <td>1.091890</td>\n",
       "      <td>2.796956</td>\n",
       "      <td>2.493385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000272398_CD24</th>\n",
       "      <td>2.578313</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.889010</td>\n",
       "      <td>1.963302</td>\n",
       "      <td>1.412966</td>\n",
       "      <td>2.786519</td>\n",
       "      <td>2.613693</td>\n",
       "      <td>2.439682</td>\n",
       "      <td>2.053146</td>\n",
       "      <td>1.702990</td>\n",
       "      <td>...</td>\n",
       "      <td>7.369764</td>\n",
       "      <td>7.730984</td>\n",
       "      <td>7.952429</td>\n",
       "      <td>7.818579</td>\n",
       "      <td>7.765602</td>\n",
       "      <td>7.552773</td>\n",
       "      <td>7.631217</td>\n",
       "      <td>7.109529</td>\n",
       "      <td>7.185068</td>\n",
       "      <td>6.756138</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11231 rows × 34256 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          21843_1#10  21843_1#100  21843_1#101  21843_1#102  \\\n",
       "ENSG00000000003_TSPAN6      5.520777     6.456208     5.878671     4.860824   \n",
       "ENSG00000000419_DPM1        5.392461     6.065923     6.838769     6.614268   \n",
       "ENSG00000000457_SCYL3       0.000174     0.352597     0.000000     0.825955   \n",
       "ENSG00000000460_C1orf112    1.471928     4.536968     4.318528     5.373009   \n",
       "ENSG00000001036_FUCA2       2.908802     3.867327     3.321747     3.736476   \n",
       "...                              ...          ...          ...          ...   \n",
       "ENSG00000271043_MTRNR2L2    8.993125    10.310311    10.112632     9.380978   \n",
       "ENSG00000271303_SRXN1       0.000000     0.000000     0.390075     0.000000   \n",
       "ENSG00000272047_GTF2H5      1.397666     1.399811     2.421133     2.250520   \n",
       "ENSG00000272325_NUDT3       0.716015     1.857059     2.499615     2.615108   \n",
       "ENSG00000272398_CD24        2.578313     0.000000     0.889010     1.963302   \n",
       "\n",
       "                          21843_1#103  21843_1#105  21843_1#106  21843_1#107  \\\n",
       "ENSG00000000003_TSPAN6       5.903640     4.513537     6.401983     5.909216   \n",
       "ENSG00000000419_DPM1         6.512403     5.527439     6.525591     6.381135   \n",
       "ENSG00000000457_SCYL3        2.201697     0.262446     0.000000     1.506837   \n",
       "ENSG00000000460_C1orf112     4.636175     4.225468     0.409785     3.668277   \n",
       "ENSG00000001036_FUCA2        4.917576     2.456866     0.577839     4.777404   \n",
       "...                               ...          ...          ...          ...   \n",
       "ENSG00000271043_MTRNR2L2     9.511378     7.531314     9.565223     9.659486   \n",
       "ENSG00000271303_SRXN1        0.000000     0.000000     0.000000     0.000000   \n",
       "ENSG00000272047_GTF2H5       2.633359     2.078498     1.513069     2.972922   \n",
       "ENSG00000272325_NUDT3        1.682406     2.295494     2.040256     1.896238   \n",
       "ENSG00000272398_CD24         1.412966     2.786519     2.613693     2.439682   \n",
       "\n",
       "                          21843_1#108  21843_1#109  ...  24539_8#88  \\\n",
       "ENSG00000000003_TSPAN6       5.366645     3.228852  ...    5.841814   \n",
       "ENSG00000000419_DPM1         6.157296     6.248478  ...    6.543807   \n",
       "ENSG00000000457_SCYL3        0.283516     3.241977  ...    0.017386   \n",
       "ENSG00000000460_C1orf112     3.057933     3.154891  ...    4.720967   \n",
       "ENSG00000001036_FUCA2        2.873857     2.536708  ...    3.070608   \n",
       "...                               ...          ...  ...         ...   \n",
       "ENSG00000271043_MTRNR2L2     7.748536    10.087547  ...    5.273407   \n",
       "ENSG00000271303_SRXN1        2.388986     0.000000  ...    0.000000   \n",
       "ENSG00000272047_GTF2H5       2.012249     1.675936  ...    2.316120   \n",
       "ENSG00000272325_NUDT3        1.236113     1.544098  ...    1.210306   \n",
       "ENSG00000272398_CD24         2.053146     1.702990  ...    7.369764   \n",
       "\n",
       "                          24539_8#89  24539_8#90  24539_8#91  24539_8#92  \\\n",
       "ENSG00000000003_TSPAN6      6.104105    6.275649    7.029407    5.806978   \n",
       "ENSG00000000419_DPM1        6.369119    7.185421    6.337047    6.162437   \n",
       "ENSG00000000457_SCYL3       0.949668    0.035526    0.032044    1.773369   \n",
       "ENSG00000000460_C1orf112    3.791536    2.696476    4.227515    4.243689   \n",
       "ENSG00000001036_FUCA2       4.491643    4.206249    3.695005    2.652845   \n",
       "...                              ...         ...         ...         ...   \n",
       "ENSG00000271043_MTRNR2L2    5.649628    5.465876    5.388675    5.636233   \n",
       "ENSG00000271303_SRXN1       3.061808    1.707139    0.000000    2.637493   \n",
       "ENSG00000272047_GTF2H5      2.299498    2.392277    1.653269    2.254969   \n",
       "ENSG00000272325_NUDT3       0.538546    1.002614    1.901335    1.037860   \n",
       "ENSG00000272398_CD24        7.730984    7.952429    7.818579    7.765602   \n",
       "\n",
       "                          24539_8#93  24539_8#94  24539_8#95  24539_8#97  \\\n",
       "ENSG00000000003_TSPAN6      6.199875    7.014180    6.228476    6.217161   \n",
       "ENSG00000000419_DPM1        5.885993    7.431358    7.013124    4.851771   \n",
       "ENSG00000000457_SCYL3       0.000000    0.108025    1.756339    2.492943   \n",
       "ENSG00000000460_C1orf112    3.227508    2.621121    3.950978    3.926914   \n",
       "ENSG00000001036_FUCA2       4.221847    3.188030    4.741496    3.872743   \n",
       "...                              ...         ...         ...         ...   \n",
       "ENSG00000271043_MTRNR2L2    1.709041    7.568684    6.345048    7.577899   \n",
       "ENSG00000271303_SRXN1       0.000000    0.000000    0.284224    0.000000   \n",
       "ENSG00000272047_GTF2H5      3.208027    4.137374    1.847985    2.144427   \n",
       "ENSG00000272325_NUDT3       2.139444    0.852408    1.091890    2.796956   \n",
       "ENSG00000272398_CD24        7.552773    7.631217    7.109529    7.185068   \n",
       "\n",
       "                          24539_8#98  \n",
       "ENSG00000000003_TSPAN6      6.034232  \n",
       "ENSG00000000419_DPM1        4.937248  \n",
       "ENSG00000000457_SCYL3       1.363441  \n",
       "ENSG00000000460_C1orf112    4.211904  \n",
       "ENSG00000001036_FUCA2       4.374577  \n",
       "...                              ...  \n",
       "ENSG00000271043_MTRNR2L2    7.012576  \n",
       "ENSG00000271303_SRXN1       0.000000  \n",
       "ENSG00000272047_GTF2H5      2.105545  \n",
       "ENSG00000272325_NUDT3       2.493385  \n",
       "ENSG00000272398_CD24        6.756138  \n",
       "\n",
       "[11231 rows x 34256 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Phenotype\n",
    "phenotype_file = input_files_dir+\"phenotype.csv.pkl\"\n",
    "phenotype = pd.read_pickle(phenotype_file)\n",
    "phenotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Phenotype shape BEFORE selection: (11231, 34256)\n",
      "Phenotype shape AFTER selection: (11231, 33964)\n"
     ]
    }
   ],
   "source": [
    "print(\"Phenotype shape BEFORE selection: {}\".format(phenotype.shape))\n",
    "phenotype = xr.DataArray(phenotype.values, dims=[\"trait\", \"cell\"], coords={\"trait\": phenotype.index.values, \"cell\": phenotype.columns.values})\n",
    "phenotype = phenotype.sel(cell=sample_mapping[\"phenotype_sample_id\"].values)\n",
    "print(\"Phenotype shape AFTER selection: {}\".format(phenotype.shape))\n",
    "assert all(phenotype.cell.values == sample_mapping[\"phenotype_sample_id\"].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><svg style=\"position: absolute; width: 0; height: 0; overflow: hidden\">\n",
       "<defs>\n",
       "<symbol id=\"icon-database\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M16 0c-8.837 0-16 2.239-16 5v4c0 2.761 7.163 5 16 5s16-2.239 16-5v-4c0-2.761-7.163-5-16-5z\"></path>\n",
       "<path d=\"M16 17c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "<path d=\"M16 26c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "</symbol>\n",
       "<symbol id=\"icon-file-text2\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M28.681 7.159c-0.694-0.947-1.662-2.053-2.724-3.116s-2.169-2.030-3.116-2.724c-1.612-1.182-2.393-1.319-2.841-1.319h-15.5c-1.378 0-2.5 1.121-2.5 2.5v27c0 1.378 1.122 2.5 2.5 2.5h23c1.378 0 2.5-1.122 2.5-2.5v-19.5c0-0.448-0.137-1.23-1.319-2.841zM24.543 5.457c0.959 0.959 1.712 1.825 2.268 2.543h-4.811v-4.811c0.718 0.556 1.584 1.309 2.543 2.268zM28 29.5c0 0.271-0.229 0.5-0.5 0.5h-23c-0.271 0-0.5-0.229-0.5-0.5v-27c0-0.271 0.229-0.5 0.5-0.5 0 0 15.499-0 15.5 0v7c0 0.552 0.448 1 1 1h7v19.5z\"></path>\n",
       "<path d=\"M23 26h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 22h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 18h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "</symbol>\n",
       "</defs>\n",
       "</svg>\n",
       "<style>/* CSS stylesheet for displaying xarray objects in jupyterlab.\n",
       " *\n",
       " */\n",
       "\n",
       ":root {\n",
       "  --xr-font-color0: var(--jp-content-font-color0, rgba(0, 0, 0, 1));\n",
       "  --xr-font-color2: var(--jp-content-font-color2, rgba(0, 0, 0, 0.54));\n",
       "  --xr-font-color3: var(--jp-content-font-color3, rgba(0, 0, 0, 0.38));\n",
       "  --xr-border-color: var(--jp-border-color2, #e0e0e0);\n",
       "  --xr-disabled-color: var(--jp-layout-color3, #bdbdbd);\n",
       "  --xr-background-color: var(--jp-layout-color0, white);\n",
       "  --xr-background-color-row-even: var(--jp-layout-color1, white);\n",
       "  --xr-background-color-row-odd: var(--jp-layout-color2, #eeeeee);\n",
       "}\n",
       "\n",
       "html[theme=dark],\n",
       "body.vscode-dark {\n",
       "  --xr-font-color0: rgba(255, 255, 255, 1);\n",
       "  --xr-font-color2: rgba(255, 255, 255, 0.54);\n",
       "  --xr-font-color3: rgba(255, 255, 255, 0.38);\n",
       "  --xr-border-color: #1F1F1F;\n",
       "  --xr-disabled-color: #515151;\n",
       "  --xr-background-color: #111111;\n",
       "  --xr-background-color-row-even: #111111;\n",
       "  --xr-background-color-row-odd: #313131;\n",
       "}\n",
       "\n",
       ".xr-wrap {\n",
       "  display: block;\n",
       "  min-width: 300px;\n",
       "  max-width: 700px;\n",
       "}\n",
       "\n",
       ".xr-text-repr-fallback {\n",
       "  /* fallback to plain text repr when CSS is not injected (untrusted notebook) */\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-header {\n",
       "  padding-top: 6px;\n",
       "  padding-bottom: 6px;\n",
       "  margin-bottom: 4px;\n",
       "  border-bottom: solid 1px var(--xr-border-color);\n",
       "}\n",
       "\n",
       ".xr-header > div,\n",
       ".xr-header > ul {\n",
       "  display: inline;\n",
       "  margin-top: 0;\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-obj-type,\n",
       ".xr-array-name {\n",
       "  margin-left: 2px;\n",
       "  margin-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-obj-type {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-sections {\n",
       "  padding-left: 0 !important;\n",
       "  display: grid;\n",
       "  grid-template-columns: 150px auto auto 1fr 20px 20px;\n",
       "}\n",
       "\n",
       ".xr-section-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-section-item input {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-section-item input + label {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label {\n",
       "  cursor: pointer;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label:hover {\n",
       "  color: var(--xr-font-color0);\n",
       "}\n",
       "\n",
       ".xr-section-summary {\n",
       "  grid-column: 1;\n",
       "  color: var(--xr-font-color2);\n",
       "  font-weight: 500;\n",
       "}\n",
       "\n",
       ".xr-section-summary > span {\n",
       "  display: inline-block;\n",
       "  padding-left: 0.5em;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in + label:before {\n",
       "  display: inline-block;\n",
       "  content: '►';\n",
       "  font-size: 11px;\n",
       "  width: 15px;\n",
       "  text-align: center;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label:before {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label:before {\n",
       "  content: '▼';\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label > span {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-section-summary,\n",
       ".xr-section-inline-details {\n",
       "  padding-top: 4px;\n",
       "  padding-bottom: 4px;\n",
       "}\n",
       "\n",
       ".xr-section-inline-details {\n",
       "  grid-column: 2 / -1;\n",
       "}\n",
       "\n",
       ".xr-section-details {\n",
       "  display: none;\n",
       "  grid-column: 1 / -1;\n",
       "  margin-bottom: 5px;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked ~ .xr-section-details {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-array-wrap {\n",
       "  grid-column: 1 / -1;\n",
       "  display: grid;\n",
       "  grid-template-columns: 20px auto;\n",
       "}\n",
       "\n",
       ".xr-array-wrap > label {\n",
       "  grid-column: 1;\n",
       "  vertical-align: top;\n",
       "}\n",
       "\n",
       ".xr-preview {\n",
       "  color: var(--xr-font-color3);\n",
       "}\n",
       "\n",
       ".xr-array-preview,\n",
       ".xr-array-data {\n",
       "  padding: 0 5px !important;\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-array-data,\n",
       ".xr-array-in:checked ~ .xr-array-preview {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-array-in:checked ~ .xr-array-data,\n",
       ".xr-array-preview {\n",
       "  display: inline-block;\n",
       "}\n",
       "\n",
       ".xr-dim-list {\n",
       "  display: inline-block !important;\n",
       "  list-style: none;\n",
       "  padding: 0 !important;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list li {\n",
       "  display: inline-block;\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list:before {\n",
       "  content: '(';\n",
       "}\n",
       "\n",
       ".xr-dim-list:after {\n",
       "  content: ')';\n",
       "}\n",
       "\n",
       ".xr-dim-list li:not(:last-child):after {\n",
       "  content: ',';\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-has-index {\n",
       "  font-weight: bold;\n",
       "}\n",
       "\n",
       ".xr-var-list,\n",
       ".xr-var-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-var-item > div,\n",
       ".xr-var-item label,\n",
       ".xr-var-item > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-even);\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-var-item > .xr-var-name:hover span {\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-var-list > li:nth-child(odd) > div,\n",
       ".xr-var-list > li:nth-child(odd) > label,\n",
       ".xr-var-list > li:nth-child(odd) > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-odd);\n",
       "}\n",
       "\n",
       ".xr-var-name {\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-var-dims {\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-var-dtype {\n",
       "  grid-column: 3;\n",
       "  text-align: right;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-var-preview {\n",
       "  grid-column: 4;\n",
       "}\n",
       "\n",
       ".xr-var-name,\n",
       ".xr-var-dims,\n",
       ".xr-var-dtype,\n",
       ".xr-preview,\n",
       ".xr-attrs dt {\n",
       "  white-space: nowrap;\n",
       "  overflow: hidden;\n",
       "  text-overflow: ellipsis;\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-var-name:hover,\n",
       ".xr-var-dims:hover,\n",
       ".xr-var-dtype:hover,\n",
       ".xr-attrs dt:hover {\n",
       "  overflow: visible;\n",
       "  width: auto;\n",
       "  z-index: 1;\n",
       "}\n",
       "\n",
       ".xr-var-attrs,\n",
       ".xr-var-data {\n",
       "  display: none;\n",
       "  background-color: var(--xr-background-color) !important;\n",
       "  padding-bottom: 5px !important;\n",
       "}\n",
       "\n",
       ".xr-var-attrs-in:checked ~ .xr-var-attrs,\n",
       ".xr-var-data-in:checked ~ .xr-var-data {\n",
       "  display: block;\n",
       "}\n",
       "\n",
       ".xr-var-data > table {\n",
       "  float: right;\n",
       "}\n",
       "\n",
       ".xr-var-name span,\n",
       ".xr-var-data,\n",
       ".xr-attrs {\n",
       "  padding-left: 25px !important;\n",
       "}\n",
       "\n",
       ".xr-attrs,\n",
       ".xr-var-attrs,\n",
       ".xr-var-data {\n",
       "  grid-column: 1 / -1;\n",
       "}\n",
       "\n",
       "dl.xr-attrs {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  display: grid;\n",
       "  grid-template-columns: 125px auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt,\n",
       ".xr-attrs dd {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  float: left;\n",
       "  padding-right: 10px;\n",
       "  width: auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt {\n",
       "  font-weight: normal;\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-attrs dt:hover span {\n",
       "  display: inline-block;\n",
       "  background: var(--xr-background-color);\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-attrs dd {\n",
       "  grid-column: 2;\n",
       "  white-space: pre-wrap;\n",
       "  word-break: break-all;\n",
       "}\n",
       "\n",
       ".xr-icon-database,\n",
       ".xr-icon-file-text2 {\n",
       "  display: inline-block;\n",
       "  vertical-align: middle;\n",
       "  width: 1em;\n",
       "  height: 1.5em !important;\n",
       "  stroke-width: 0;\n",
       "  stroke: currentColor;\n",
       "  fill: currentColor;\n",
       "}\n",
       "</style><pre class='xr-text-repr-fallback'>&lt;xarray.DataArray (trait: 11231, cell: 33964)&gt;\n",
       "array([[5.52077706e+00, 6.45620778e+00, 5.87867097e+00, ...,\n",
       "        6.22847584e+00, 6.21716111e+00, 6.03423233e+00],\n",
       "       [5.39246051e+00, 6.06592255e+00, 6.83876922e+00, ...,\n",
       "        7.01312448e+00, 4.85177110e+00, 4.93724790e+00],\n",
       "       [1.74155604e-04, 3.52596612e-01, 0.00000000e+00, ...,\n",
       "        1.75633852e+00, 2.49294278e+00, 1.36344116e+00],\n",
       "       ...,\n",
       "       [1.39766627e+00, 1.39981065e+00, 2.42113299e+00, ...,\n",
       "        1.84798489e+00, 2.14442714e+00, 2.10554501e+00],\n",
       "       [7.16014880e-01, 1.85705932e+00, 2.49961503e+00, ...,\n",
       "        1.09189033e+00, 2.79695639e+00, 2.49338533e+00],\n",
       "       [2.57831281e+00, 0.00000000e+00, 8.89010394e-01, ...,\n",
       "        7.10952922e+00, 7.18506822e+00, 6.75613759e+00]])\n",
       "Coordinates:\n",
       "  * trait    (trait) object &#x27;ENSG00000000003_TSPAN6&#x27; ... &#x27;ENSG00000272398_CD24&#x27;\n",
       "  * cell     (cell) object &#x27;21843_1#10&#x27; &#x27;21843_1#100&#x27; ... &#x27;24539_8#98&#x27;</pre><div class='xr-wrap' hidden><div class='xr-header'><div class='xr-obj-type'>xarray.DataArray</div><div class='xr-array-name'></div><ul class='xr-dim-list'><li><span class='xr-has-index'>trait</span>: 11231</li><li><span class='xr-has-index'>cell</span>: 33964</li></ul></div><ul class='xr-sections'><li class='xr-section-item'><div class='xr-array-wrap'><input id='section-8f0cead2-e4ab-4b34-8b57-f09dc9938674' class='xr-array-in' type='checkbox' checked><label for='section-8f0cead2-e4ab-4b34-8b57-f09dc9938674' title='Show/hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-array-preview xr-preview'><span>5.521 6.456 5.879 4.861 5.904 4.514 ... 7.553 7.631 7.11 7.185 6.756</span></div><div class='xr-array-data'><pre>array([[5.52077706e+00, 6.45620778e+00, 5.87867097e+00, ...,\n",
       "        6.22847584e+00, 6.21716111e+00, 6.03423233e+00],\n",
       "       [5.39246051e+00, 6.06592255e+00, 6.83876922e+00, ...,\n",
       "        7.01312448e+00, 4.85177110e+00, 4.93724790e+00],\n",
       "       [1.74155604e-04, 3.52596612e-01, 0.00000000e+00, ...,\n",
       "        1.75633852e+00, 2.49294278e+00, 1.36344116e+00],\n",
       "       ...,\n",
       "       [1.39766627e+00, 1.39981065e+00, 2.42113299e+00, ...,\n",
       "        1.84798489e+00, 2.14442714e+00, 2.10554501e+00],\n",
       "       [7.16014880e-01, 1.85705932e+00, 2.49961503e+00, ...,\n",
       "        1.09189033e+00, 2.79695639e+00, 2.49338533e+00],\n",
       "       [2.57831281e+00, 0.00000000e+00, 8.89010394e-01, ...,\n",
       "        7.10952922e+00, 7.18506822e+00, 6.75613759e+00]])</pre></div></div></li><li class='xr-section-item'><input id='section-64d2ddc1-ea27-45ac-b1bf-d604527ba7a1' class='xr-section-summary-in' type='checkbox'  checked><label for='section-64d2ddc1-ea27-45ac-b1bf-d604527ba7a1' class='xr-section-summary' >Coordinates: <span>(2)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'><li class='xr-var-item'><div class='xr-var-name'><span class='xr-has-index'>trait</span></div><div class='xr-var-dims'>(trait)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;ENSG00000000003_TSPAN6&#x27; ... &#x27;EN...</div><input id='attrs-5ca9a559-cb1b-4fa4-9b50-d55be1d437bb' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-5ca9a559-cb1b-4fa4-9b50-d55be1d437bb' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-06c9796e-75ae-4b30-b0a5-82c5527d5ae5' class='xr-var-data-in' type='checkbox'><label for='data-06c9796e-75ae-4b30-b0a5-82c5527d5ae5' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;ENSG00000000003_TSPAN6&#x27;, &#x27;ENSG00000000419_DPM1&#x27;,\n",
       "       &#x27;ENSG00000000457_SCYL3&#x27;, ..., &#x27;ENSG00000272047_GTF2H5&#x27;,\n",
       "       &#x27;ENSG00000272325_NUDT3&#x27;, &#x27;ENSG00000272398_CD24&#x27;], dtype=object)</pre></div></li><li class='xr-var-item'><div class='xr-var-name'><span class='xr-has-index'>cell</span></div><div class='xr-var-dims'>(cell)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>&#x27;21843_1#10&#x27; ... &#x27;24539_8#98&#x27;</div><input id='attrs-7f24239c-8e87-4749-9c41-ab20d56363e0' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-7f24239c-8e87-4749-9c41-ab20d56363e0' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-dfc4a039-aad8-48ac-98a5-07b25e59bbf0' class='xr-var-data-in' type='checkbox'><label for='data-dfc4a039-aad8-48ac-98a5-07b25e59bbf0' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><pre>array([&#x27;21843_1#10&#x27;, &#x27;21843_1#100&#x27;, &#x27;21843_1#101&#x27;, ..., &#x27;24539_8#95&#x27;,\n",
       "       &#x27;24539_8#97&#x27;, &#x27;24539_8#98&#x27;], dtype=object)</pre></div></li></ul></div></li><li class='xr-section-item'><input id='section-011065f0-70fb-4c56-85ff-fb25d44b35c5' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-011065f0-70fb-4c56-85ff-fb25d44b35c5' class='xr-section-summary'  title='Expand/collapse section'>Attributes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><dl class='xr-attrs'></dl></div></li></ul></div></div>"
      ],
      "text/plain": [
       "<xarray.DataArray (trait: 11231, cell: 33964)>\n",
       "array([[5.52077706e+00, 6.45620778e+00, 5.87867097e+00, ...,\n",
       "        6.22847584e+00, 6.21716111e+00, 6.03423233e+00],\n",
       "       [5.39246051e+00, 6.06592255e+00, 6.83876922e+00, ...,\n",
       "        7.01312448e+00, 4.85177110e+00, 4.93724790e+00],\n",
       "       [1.74155604e-04, 3.52596612e-01, 0.00000000e+00, ...,\n",
       "        1.75633852e+00, 2.49294278e+00, 1.36344116e+00],\n",
       "       ...,\n",
       "       [1.39766627e+00, 1.39981065e+00, 2.42113299e+00, ...,\n",
       "        1.84798489e+00, 2.14442714e+00, 2.10554501e+00],\n",
       "       [7.16014880e-01, 1.85705932e+00, 2.49961503e+00, ...,\n",
       "        1.09189033e+00, 2.79695639e+00, 2.49338533e+00],\n",
       "       [2.57831281e+00, 0.00000000e+00, 8.89010394e-01, ...,\n",
       "        7.10952922e+00, 7.18506822e+00, 6.75613759e+00]])\n",
       "Coordinates:\n",
       "  * trait    (trait) object 'ENSG00000000003_TSPAN6' ... 'ENSG00000272398_CD24'\n",
       "  * cell     (cell) object '21843_1#10' '21843_1#100' ... '24539_8#98'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phenotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter on specific gene-SNP pairs\n",
    "# eQTL from endodiff (ips+mesendo+defendo)\n",
    "endo_eqtl_file = \"/hps/nobackup/stegle/users/acuomo/all_scripts/struct_LMM2/sc_endodiff/eqtl_allstages.tsv\"\n",
    "endo_eqtl = pd.read_csv(endo_eqtl_file, sep=\"\\t\", index_col = False)\n",
    "endo_eqtl[\"chrom\"] = [int(i[:i.find(\"_\")]) for i in endo_eqtl[\"snp_id\"]]\n",
    "genes = endo_eqtl[endo_eqtl['chrom']==int(chrom)]['feature'].unique()\n",
    "# genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "124"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up model\n",
    "n_samples = phenotype.shape[1]\n",
    "M = ones((n_samples, 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ENSG00000147813_NAPRT1'"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# i = 2\n",
    "# trait_name = genes[i]\n",
    "trait_name = \"ENSG00000147813_NAPRT1\"\n",
    "trait_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "y = phenotype.sel(trait=trait_name)\n",
    "y = quantile_gaussianize(y)\n",
    "E = quantile_gaussianize(E)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "y = np.asarray(y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(y == 0).astype(int).sum()/len(y)>0.3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.matshow(E @ E.T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.hist(y)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "slmm2 = StructLMM2(y, M, E, Ls)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "leads = endo_eqtl[endo_eqtl['feature']==trait_name]['snp_id'].unique()\n",
    "G_tmp = G_exp[:,G_exp['snp'].isin(leads)]\n",
    "G_tmp.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "G_maf = G_sel[G_sel['sample'].isin(donors),G_sel['snp'].isin(leads)]\n",
    "G_maf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import division\n",
    "def compute_maf(X):\n",
    "    r\"\"\"Compute minor allele frequencies.\n",
    "    It assumes that ``X`` encodes 0, 1, and 2 representing the number\n",
    "    of alleles (or dosage), or ``NaN`` to represent missing values.\n",
    "    Parameters\n",
    "    ----------\n",
    "    X : array_like\n",
    "        Genotype matrix.\n",
    "    Returns\n",
    "    -------\n",
    "    array_like\n",
    "        Minor allele frequencies.\n",
    "    Examples\n",
    "    --------\n",
    "    .. doctest::\n",
    "        >>> from numpy.random import RandomState\n",
    "        >>> from limix.qc import compute_maf\n",
    "        >>>\n",
    "        >>> random = RandomState(0)\n",
    "        >>> X = random.randint(0, 3, size=(100, 10))\n",
    "        >>>\n",
    "        >>> print(compute_maf(X)) # doctest: +FLOAT_CMP\n",
    "        [0.49  0.49  0.445 0.495 0.5   0.45  0.48  0.48  0.47  0.435]\n",
    "    \"\"\"\n",
    "    import dask.array as da\n",
    "    import xarray as xr\n",
    "    from pandas import DataFrame\n",
    "    from numpy import isnan, logical_not, minimum, nansum\n",
    "    if isinstance(X, da.Array):\n",
    "        s0 = da.nansum(X, axis=0).compute()\n",
    "        denom = 2 * (X.shape[0] - da.isnan(X).sum(axis=0)).compute()\n",
    "    elif isinstance(X, DataFrame):\n",
    "        s0 = X.sum(axis=0, skipna=True)\n",
    "        denom = 2 * logical_not(X.isna()).sum(axis=0)\n",
    "    elif isinstance(X, xr.DataArray):\n",
    "        if \"sample\" in X.dims:\n",
    "            kwargs = {\"dim\": \"sample\"}\n",
    "        else:\n",
    "            kwargs = {\"axis\": 0}\n",
    "        s0 = X.sum(skipna=True, **kwargs)\n",
    "        denom = 2 * logical_not(isnan(X)).sum(**kwargs)\n",
    "    else:\n",
    "        s0 = nansum(X, axis=0)\n",
    "        denom = 2 * logical_not(isnan(X)).sum(axis=0)\n",
    "    s0 = s0 / denom\n",
    "    s1 = 1 - s0\n",
    "    maf = minimum(s0, s1)\n",
    "    if hasattr(maf, \"name\"):\n",
    "        maf.name = \"maf\"\n",
    "    return maf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mafs = compute_maf(G_maf).compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mafs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "b = slmm2.predict_interaction(G_tmp, mafs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# beta_G value for each of the SNPs tested\n",
    "beta_G = b[0]\n",
    "beta_G.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# beta_GxE values (should be n_cells x nSNPs)\n",
    "beta_GxE = b[1][0]\n",
    "beta_GxE.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cells = phenotype[\"cell\"].values\n",
    "snps = G_tmp[\"variant\"].values\n",
    "\n",
    "betas_df = pd.DataFrame(data = beta_GxE, columns = snps, index = cells)\n",
    "betas_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "folder = \"/hps/nobackup/stegle/users/acuomo/all_scripts/struct_LMM2/sc_endodiff/new/PCA/\"\n",
    "outfilename0 = f\"{folder}beta_GxE/PCA{n_factors}/{trait_name}.tsv\"\n",
    "outfilename0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# betas_df.to_csv(outfilename0, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "betaG_df = pd.DataFrame(data = beta_G, index = snps, columns = ['beta_G'])\n",
    "betaG_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "outfilename1 = f\"{folder}beta_G/PCA{n_factors}/{trait_name}.tsv\"\n",
    "outfilename1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# betaG_df.to_csv(outfilename1, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "from math import ceil\n",
    "import matplotlib.pyplot as plt\n",
    "##\n",
    "beta = np.array([])\n",
    "for i in range(beta_GxE.shape[1]):\n",
    "# for i in range(5):\n",
    "    # get single-cell effect sizes\n",
    "    v = beta_GxE[:,i] \n",
    "    # get min and max\n",
    "    ymin = min(v)\n",
    "    ymax = max(v)\n",
    "    # rank values\n",
    "    order = v.argsort()\n",
    "    y = v[order]\n",
    "    x = range(len(v))\n",
    "    plt.scatter(x,y)\n",
    "    # define 10% of cells\n",
    "    xx = ceil(0.1*len(v))\n",
    "    plt.vlines(x=xx,ymin=ymin, ymax=ymax)\n",
    "    plt.hlines(y=y[xx],xmin=0, xmax=len(v))\n",
    "    plt.vlines(x=len(v)-xx,ymin=ymin, ymax=ymax)\n",
    "    plt.hlines(y=y[len(v)-xx],xmin=0, xmax=len(v))\n",
    "    # add title\n",
    "#     plt.title(df.columns.values[i])\n",
    "#     plt.title(str(i))\n",
    "    plt.show()\n",
    "    beta = np.append(beta,y[len(v)-xx]-y[xx])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "beta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "E.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "E_file = input_files_dir+\"10PCs.csv\"\n",
    "E = pd.read_csv(E_file, index_col = 0)\n",
    "E = xr.DataArray(E.values, dims=[\"cell\", \"pc\"], coords={\"cell\": E.index.values, \"pc\": E.columns.values})\n",
    "E = E.sel(cell=sample_mapping[\"phenotype_sample_id\"].values)\n",
    "assert all(E.cell.values == sample_mapping[\"phenotype_sample_id\"].values)\n",
    "# E = quantile_gaussianize(E)\n",
    "plt.scatter(E.values[:,0],E.values[:,1], c = beta_GxE, alpha = 0.6)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (struct-lmm2)",
   "language": "python",
   "name": "struct-lmm2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
